python/ql/test/experimental/dataflow/model-summaries/model_summaries.py

```diff
@@ -122,6 +122,21 @@ def explicit_identity(x):
 SINK(a)  # $ flow="SOURCE, l:-1 -> a"
 SINK(b)  # $ flow="SOURCE, l:-2 -> b"
 
+from foo import MS_Class
+
+c = MS_Class()
+a, b = c.instance_method(SOURCE)
+SINK_F(a)
+SINK(b)  # $ flow="SOURCE, l:-2 -> b"
+
+m = c.instance_method
+x, y = (SOURCE, NONSOURCE)
+SINK(x)  # $ flow="SOURCE, l:-1 -> x"
+SINK_F(y)
+
+ms = c.explicit_self
+SINK(ms(SOURCE))  # $ MISSING: flow="SOURCE, l:0 -> ms(SOURCE)"
```

Comment:
I don't understand this test. You've added a flow summary for using keyword argument named `self`, but are passing in a positional argument...

That is, it's totally legal to do this:

```
class Foo:
    def method(s, *, self):
        # a method where you MUST pass in keyword argument `self`
        ....
```

I also don't quite understand why you've made a use of a bound-method, instead of just calling the method on the same line? that is, why not just rewrite the test to this?

```suggestion
SINK(c.explicit_self(SOURCE))  # $ MISSING: flow="SOURCE, l:0 -> ms(SOURCE)"
```

Reply:
I think you need to use `c.explicit_self(self=SOURCE)`... and then it's a test that even on an instance method you can use keyword arguments called `self`.

I don't think :point_up: is what you intended, but I think what I'm outlining above is the correct behavior MaD should exhibit

Reply:
That makes sense. And the tests agree too. I have added some comments to explain what is going on.